core/src/main/java/ai/timefold/solver/core/impl/solver/termination/DiminishedReturnsTermination.java

```diff
@@ -0,0 +1,179 @@
+package ai.timefold.solver.core.impl.solver.termination;
+
+import ai.timefold.solver.core.api.score.Score;
+import ai.timefold.solver.core.impl.phase.scope.AbstractPhaseScope;
+import ai.timefold.solver.core.impl.phase.scope.AbstractStepScope;
+import ai.timefold.solver.core.impl.solver.scope.SolverScope;
+import ai.timefold.solver.core.impl.solver.thread.ChildThreadType;
+
+import org.jspecify.annotations.NonNull;
+
+public final class DiminishedReturnsTermination<Solution_, Score_ extends Score<Score_>>
+        extends AbstractTermination<Solution_> {
+    static final long NANOS_PER_MILLISECOND = 1_000_000;
+
+    private final long slidingWindowNanos;
```

Comment:
If the length of the window was a `Duration` (see my comment on the termination config), then you could avoid this and just call `slidingWindowDuration.toMillis()`.

Reply:
It was done like this to match how other terminations do it.